pkg/util/api.go

```diff
@@ -72,3 +76,14 @@ func WaitForAPIServerReady(ctx context.Context, client clientset.Interface, time
 
 	return nil
 }
+
+func BuildControllerEventRecorder(k8s clientset.Interface, controllerName string, nodeName string) record.EventRecorder {
+	logrus.Infof("Creating %s event broadcaster", controllerName)
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(logrus.Infof)
+	eventBroadcaster.StartRecordingToSink(&coregetter.EventSinkImpl{Interface: k8s.CoreV1().Events(metav1.NamespaceSystem)})
+	if nodeName != "" {
+		return eventBroadcaster.NewRecorder(schemes.All, v1.EventSource{Component: controllerName, Host: nodeName})
+	}
+	return eventBroadcaster.NewRecorder(schemes.All, v1.EventSource{Component: controllerName})
```

Comment:
It's a string field, not setting it is the same as setting it to an empty string.
```suggestion
	return eventBroadcaster.NewRecorder(schemes.All, v1.EventSource{Component: controllerName, Host: nodeName})
```

Reply:
fixed